In [1]:
from sympy import symbols,solve
from sympy.physics.units import m,newton,force,kilo,Quantity
kilonewton = kilonewtons = kN = Quantity("kilonewton", force, kilo*newton, "kN")

## Free Body Diagram of Part b-c-d-e

![FBD 1](../../images/sdbeams/fbd/drawing-2.svg)

In [2]:
Hb,Vb,Vd = symbols('Hb Vb Vd')

### $\sum M_b = 0~~~~~+ccw$

In [3]:
sum_Mb = -65*kN*4*m  - 28*kN/m*(6*m+2*m)*(6*m+2*m)/2 + Vd*6*m
sum_Mb

-1156*kilonewton*meter + 6*meter*Vd

In [4]:
Vd = solve(sum_Mb,Vd)[0].n()
Vd

192.666666666667*kilonewton

### $\sum F_y = 0~~~~~+\uparrow$

In [5]:
sum_Fy = Vb - 28*kN/m*(6*m+2*m) - 65*kN + Vd
sum_Fy

-96.3333333333333*kilonewton + Vb

In [6]:
Vb = solve(sum_Fy,Vb)[0].n()
Vb

96.3333333333333*kilonewton

### $\sum F_x = 0~~~~~+ \rightarrow$

In [7]:
sum_Fx = Hb
Hb = solve(sum_Fx,Hb)[0].n()
Hb

0

In [8]:
Vd/kN, Vb/kN, Hb/kN

(192.666666666667, 96.3333333333333, 0)

## Free Body Diagram of part a-b
![FBD a-b](../../images/sdbeams/fbd/drawing-3.svg)

In [9]:
Ha,Ma,Va = symbols('Ha Ma Va')

### $\sum F_x = 0~~~~~+\rightarrow$

In [10]:
sum_Fx = Ha - Hb
Ha = solve(sum_Fx,Ha)[0].n()
Ha

0

### $\sum F_y = 0~~~~~+\uparrow$

In [11]:
sum_Fy = Va - 28*(kN/m)*8*m - Vb
Va = solve(sum_Fy,Va)[0].n()
Va

320.333333333333*kilonewton

### $\sum M_a = 0~~~~~+ ccw$

In [12]:
sum_Ma = Ma - 28*(kN/m)*8*m*8*m/2 - Vb*8*m
Ma = solve(sum_Ma,Ma)[0].n()
Ma

1666.66666666667*kilonewton*meter

## Equilibrium Check

![Entire Beam FBD](../../images/sdbeams/fbd/drawing-4.svg)

### $\sum F_x~~~~~+ \rightarrow$

In [13]:
0*kN

0

### $\sum F_y~~~~~+ \uparrow$

In [14]:
320.3*kN + 192.7*kN - 28*(kN/m)*(14*m+2*m) - 65*kN

0

### $\sum M_e~~~~~+ccw$

In [15]:
1667*kN*m + 28*(kN/m)*(14*m+2*m)*(14*m+2*m)/2 + 65*kN*(2*m+2*m) - 320.3*kN*(14*m+2*m) - 192.7*kN*2*m

0.799999999999841*kilonewton*meter

This is 0.8 kN-m which is very small compared to the moment at *a* (1667 kN-m) and is due entirely to floating point round-off errors.  This is further demonstrated below where we use the computed (and non-rounded) values.

In [16]:
Ma + 28*(kN/m)*(14*m+2*m)*(14*m+2*m)/2 + 65*kN*(2*m+2*m) - Va*(14*m+2*m) - Vd*2*m

6.25277607468888e-13*kilonewton*meter

$6.25 \times 10^{-13}$ is easily close enough to zero, attributable to normal floating point truncation errors.